domain/secret/state/state_test.go

```diff
@@ -3643,3 +3643,86 @@ func (s *stateSuite) TestGetObsoleteUserSecretRevisionsReadyToPrune(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, jc.SameContents, []string{uri.ID + "/1"})
 }
+
+func (s *stateSuite) TestChangeSecretBackend(c *gc.C) {
+	st := newSecretState(c, s.TxnRunnerFactory())
+	ctx := context.Background()
+
+	s.setupUnits(c, "mysql")
+	uriCharm1 := coresecrets.NewURI()
```

Comment:
Do we need Charm1 and Charm2, and User1 and User2?
Would just a one charm secret and one user secret suffice?